executor/ddl.go

```diff
@@ -463,14 +463,14 @@ func (e *DDLExec) getRecoverTableByJobID(s *ast.RecoverTableStmt, t *meta.Meta,
 // it will use the `start_ts` of DDL job as snapshot to get the dropped/truncated table information.
 func GetDropOrTruncateTableInfoFromJobs(jobs []*model.Job, gcSafePoint uint64, dom *domain.Domain, fn func(*model.Job, *model.TableInfo) (bool, error)) (bool, error) {
 	for _, job := range jobs {
-		if job.Type != model.ActionDropTable && job.Type != model.ActionTruncateTable {
-			continue
-		}
 		// Check GC safe point for getting snapshot infoSchema.
 		err := gcutil.ValidateSnapshotWithGCSafePoint(job.StartTS, gcSafePoint)
```

Comment:
for older history jobs, it will smaller than gc safepoint, it could be `continue` here (rather than error)? cause there still some newer history jobs  bigger than gc safe point which is our target.

Reply:
If there is one DDL job which `startTS < gcSafePoint`, the other dropped/truncate DDL jobs are not needed?

Reply:
The iteration was from the latest history DDL job to the oldest history DDL job.

Reply:
ok